### PR TITLE
Disposals Fix

### PR DESCRIPTION
**Pull request by ThatDan123**
-Fixes disposals outlet error

-Fixes disposals outlet sound

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs [hunk 1]
```diff
@@ -169,8 +169,11 @@ private void ThrowContainedItems(Vector3 throwVector)
 		{
 			foreach (ObjectBehaviour item in containedItems)
 			{
+				if(item == null) continue;
 				ThrowItem(item, throwVector);
 			}
+
+			containedItems.Clear();
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs]: Should be cleared in `EjectContentsAndThrow()`

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs [hunk 2]
```diff
@@ -169,8 +169,11 @@ private void ThrowContainedItems(Vector3 throwVector)
 		{
 			foreach (ObjectBehaviour item in containedItems)
 			{
+				if(item == null) continue;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs:172]: Any idea why it is null? Despawned during traversal?

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs:172]: ~~So it becomes null after being ejected but before thrown? That's odd?~~

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs:172]: > Any idea why it is null? Despawned during traversal?

Yeah thats what im guessing

Was nulling on this line, throw vector wouldnt be null so must be the item

Vector3 vector = item.transform.rotation * throwVector;

**Comment by NoooneyDude** [comment]
Just as an aside, would be nice to see Inventory reworked or a new system to handle objects, for things like closets, lockers, morgues, dna scanners, disposal containers and the like.

One particular offender is DNA scanners being very intertwined with lockers; ideally they would just share a new component.

Yeah, probs best a new component as opposed to modifying Inventory.

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Disposals/DisposalVirtualContainer.cs]: ah yeah missed that

**Comment by NoooneyDude** [comment]
Speeeeed merge

**Comment by PerfectTangent** [comment]
> Just as an aside, would be nice to see Inventory reworked or a new system to handle objects, for things like closets, lockers, morgues, dna scanners, disposal containers and the like.
> 
> One particular offender is DNA scanners being very intertwined with lockers; ideally they would just share a new component.
> 
> Yeah, probs best a new component as opposed to modifying Inventory.

Mak issu

**Comment by PerfectTangent** [comment]
did this fix #6568?

**Comment by ThatDan123** [comment]
> did this fix #6568?

Nope

**Pull request merged**